### PR TITLE
Update to reflect current Trio CI practice

### PR DESCRIPTION
**Pull request by oremanj**
* Remove support for Python 3.5
* Use Github Actions where we can and Travis where we must - no more AppVeyor
* Add infrastructure for mypy type checking
* Check formatting with black, not yapf

This should tide us over until we have more full-featured CI infrastructure automation tools.

### CODE REVIEW

**Comment by oremanj** [comment]
I didn't add automated tests, but I did test this manually by creating a new GitHub repository, a PR against it, and making sure the CI was happy with the outcome.

File: {{cookiecutter.project_slug}}/check.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# Autoformatter *first*, to avoid double-reporting errors
+black --check setup.py {{cookiecutter.package_name}} \
+    || EXIT_STATUS=$?
+
+# Run flake8 without pycodestyle and import-related errors
+flake8 {{cookiecutter.package_name}}/ \
+    --ignore=D,E,W,F401,F403,F405,F821,F822\
+    || EXIT_STATUS=$?
+
+# Uncomment to run mypy (to check static typing)
```
Comment (pquentin) [inline {{cookiecutter.project_slug}}/check.sh:16]: We may want to mention that tests-requirements.txt also needs an uncomment?

Comment (pquentin) [reply {{cookiecutter.project_slug}}/check.sh:16]: (Feel free to merge with or without adressing this.)

Comment (oremanj) [reply {{cookiecutter.project_slug}}/check.sh:16]: `CHEATSHEET` has the whole list of things to touch if you want mypy.

**Pull request merged**